components/tikv_util/src/sys/cgroup.rs

```diff
@@ -1,520 +1,341 @@
+        let _id = iter.next().unwrap();
+        let systems = iter.next().unwrap();
+        let path = iter.next().unwrap();
+        for system in systems.split(',') {
+            subsystems.insert(system.to_owned(), path.to_owned());
         }
-
-        Ok(cpuset)
     }
+    subsystems
 }
 
-#[derive(Debug)]
-pub struct CGroupSys {
-    // subsystem name -> CGroup
-    pub cgroups: HashMap<String, CGroup>,
+fn parse_proc_cgroup_v2(lines: &str) -> HashMap<String, String> {
+    let subsystems = parse_proc_cgroup_v1(lines);
+    assert_eq!(subsystems.len(), 1);
+    subsystems
 }
 
-impl Default for CGroupSys {
-    fn default() -> Self {
-        Self::new(CGROUP_PATH, CGROUP_MOUNTINFO)
+fn parse_memory_max(line: &str) -> i64 {
+    if line == "max" {
+        return -1;
     }
+    line.parse().unwrap()
 }
 
-impl CGroupSys {
-    pub fn new(cgroup_path: &str, mountinfo_path: &str) -> CGroupSys {
-        // Parse subsystem from cgroup_path.
-        let mut subsystems: HashMap<String, CGroupSubsys> = HashMap::new();
-        let _ = file_scanner(cgroup_path, |line| {
-            if let Ok(s) = parse_subsys_from_line(line) {
-                for subsys in &s.sub_systems {
-                    subsystems.insert(subsys.to_string(), s.clone());
-                }
-            }
-        });
-
-        // Parse mount points from mountinfo_path and collect subsystems path.
-        let mut cgroups: HashMap<String, CGroup> = HashMap::new();
-        let _ = file_scanner(mountinfo_path, |line| {
-            if let Ok(mp) = parse_mount_point_from_line(line) {
-                if mp.fs_type == CGROUP_FSTYPE {
-                    for op in &mp.super_options {
-                        if let Some(sub) = subsystems.get(op) {
-                            if let Ok(path) = mp.translate(&sub.name) {
-                                cgroups.insert(op.to_string(), CGroup::new(path));
-                            }
-                        }
-                    }
-                }
-            }
-        });
-
-        Self { cgroups }
-    }
-
-    pub fn cpu_cores_quota(&self) -> Option<f64> {
-        if let Some(sub_cpu) = self.cgroups.get(CPU_SUBSYS) {
-            if let Ok(quota) = sub_cpu.read_num(CPU_QUOTA) {
-                if quota < 0 {
-                    return None;
-                }
-                if let Ok(period) = sub_cpu.read_num(CPU_PERIOD) {
-                    return Some(quota as f64 / period as f64);
-                }
+fn parse_cpu_cores(value: &str) -> HashSet<usize> {
+    let mut cores = HashSet::new();
+    for v in value.split(',') {
+        if v.contains('-') {
+            let mut v = v.split('-');
+            let s = v.next().unwrap().parse::<usize>().unwrap();
+            let e = v.next().unwrap().parse::<usize>().unwrap();
+            for x in s..=e {
+                cores.insert(x);
             }
+        } else {
+            let s = v.parse::<usize>().unwrap();
```

Comment:
I test this PR on my own machine and it panics here. It is possible that `cpuset.cpus` is just an empty file.